SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -11,38 +11,40 @@ Buffer {
 	*initClass { serverCaches = IdentityDictionary.new }
 
 	// doesn't send
-	*new { arg server, numFrames, numChannels, bufnum;
+	*new { arg server, numFrames, numChannels, bufnum, sampleRate;
 		server = server ? Server.default;
 		bufnum ?? { bufnum = server.nextBufferNumber(1) };
 		^super.newCopyArgs(
 			server,
 			bufnum,
 			numFrames,
-			numChannels
-		).sampleRate_(server.sampleRate).cache
+			numChannels,
+			sampleRate !? _.asFloat ?? 0.0
+		).cache
 	}
 
-	*alloc { arg server, numFrames, numChannels = 1, completionMessage, bufnum;
+	*alloc { arg server, numFrames, numChannels = 1, completionMessage, bufnum, sampleRate;
 		server = server ? Server.default;
 		bufnum ?? { bufnum = server.nextBufferNumber(1) };
 		^super.newCopyArgs(
 			server,
 			bufnum,
 			numFrames,
-			numChannels
-		).alloc(completionMessage).sampleRate_(server.sampleRate).cache
+			numChannels,
+			sampleRate !? _.asFloat ?? 0.0
```

Comment:
forward `sampleRate` as is! `alloc` is responsible for sending the appropriate value to the Server, see my comment in `allocMsg`.